lua/null-ls/builtins/diagnostics.lua

```diff
@@ -201,128 +201,157 @@ M.chktex = h.make_builtin({
-        args = { "--text=$TEXT", "--parse" },
+        command = "clang-tidy",
+        args = { "--quiet", "$FILENAME" },
+        to_stdin = false,
+        to_temp_file = true,
+        format = "line",
+        on_output = from_pattern(
+            "(%d+):(%d+): (%w+): (.*) %[(.*)%]", --
+            { "row", "col", "severity", "message", "code" }
+        ),
+    },
+    factory = h.generator_factory,
+})
+
+M.eslint = h.make_builtin({
+    method = DIAGNOSTICS,
+    filetypes = { "javascript", "javascriptreact", "typescript", "typescriptreact", "vue" },
+    generator_opts = {
+        command = "eslint",
+        args = { "-f", "json", "--stdin", "--stdin-filename", "$FILENAME" },
+        to_stdin = true,
+        format = "json_raw",
+        check_exit_code = function(code)
+            return code <= 1
+        end,
+        use_cache = true,
+        on_output = function(params)
+            params.messages = params.output and params.output[1] and params.output[1].messages or {}
+            if params.err then
+                table.insert(params.messages, { message = params.err })
+            end
+
+            local parser = from_json({
+                attributes = {
+                    severity = "severity",
+                },
+                severities = {
+                    default_severities["warning"],
+                    default_severities["error"],
+                },
+            })
+
+            return parser({ output = params.messages })
+        end,
+    },
+    factory = h.generator_factory,
+})
+
+M.flake8 = h.make_builtin({
+    method = DIAGNOSTICS,
+    filetypes = { "python" },
+    generator_opts = {
+        command = "flake8",
+        to_stdin = true,
+        to_stderr = true,
+        args = { "--stdin-display-name", "$FILENAME", "-" },
         format = "line",
         check_exit_code = function(code)
             return code == 0 or code == 255
         end,
         on_output = from_pattern(
-            [[(%d+):(%d+):("([%w%s]+)".*)]], --
-            { "row", "col", "message", "_quote" },
+            [[:(%d+):(%d+): (([EFW])%w+) (.*)]], --
+            { "row", "col", "code", "severity", "message" },
             {
-                adapters = { diagnostic_adapters.end_col.from_quote },
-                offsets = { col = 1, end_col = 1 },
+                severities = {
+                    E = default_severities["error"],
+                    W = default_severities["warning"],
+                    F = default_severities["information"],
+                },
             }
         ),
     },
     factory = h.generator_factory,
 })
 
-M.markdownlint = h.make_builtin({
+M.golangci_lint = h.make_builtin({
     method = DIAGNOSTICS,
-    filetypes = { "markdown" },
+    filetypes = { "go" },
     generator_opts = {
-        command = "markdownlint",
-        args = { "--stdin" },
+        command = "golangci-lint",
+        args = { "run" },
         to_stdin = true,
```

Comment:
```suggestion
        to_stderr = true,
```
I understand that we need to use `to_stderr` for linters that return non-zero return codes. In my testing linting failed unless `to_stderr` was used. 
`to_stdin` hadn't had much effect, as `golangci-lint` doesn't support receiving data via stdin, it just wants filanames/package names.

Reply:
I'll just mention that strictly speaking, `to_stderr` is independent from the exit code – it depends on the behavior of the linter itself (e.g. some linters will only output to `stderr` if linting itself failed but use exit codes to indicate results).

Reply:
If I run `cat cmd/main.go | golangci-lint run 2> /dev/null` I still get diagnostics from stdin, am I missing smth ?

Reply:
Or I'm missing something 😄 
I suggested this null-ls because it fails without this option:
```
Error executing vim.schedule lua callback: ...e/pack/packer/start/null-ls.nvim/lua/nul
l-ls/helpers.lua:135: error in generator output: cmd/main.go:12:1: commentFormatting:
put a space between `//` and comment text (gocritic)
//bad comment formatting
...
```
Happens after I reinstall neovim and lunarvim with different versions, as well. So maybe there's a bug somewhere else and this is just the workaround? 😄 
Does it work for you without `to_stderr = true`?

Reply:
No, I need to make the change but I don't understand why, any idea @jose-elias-alvarez